node/pkg/watchers/aptos/watcher.go

```diff
@@ -0,0 +1,332 @@
+package aptos
+
+import (
+	"context"
+	"encoding/binary"
+	"encoding/hex"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"time"
+
+	"github.com/certusone/wormhole/node/pkg/common"
+	"github.com/certusone/wormhole/node/pkg/p2p"
+	gossipv1 "github.com/certusone/wormhole/node/pkg/proto/gossip/v1"
+	"github.com/certusone/wormhole/node/pkg/readiness"
+	"github.com/certusone/wormhole/node/pkg/supervisor"
+	eth_common "github.com/ethereum/go-ethereum/common"
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/prometheus/client_golang/prometheus/promauto"
+	"github.com/tidwall/gjson"
+	"github.com/wormhole-foundation/wormhole/sdk/vaa"
+	"go.uber.org/zap"
+)
+
+type (
+	// Watcher is responsible for looking over Aptos blockchain and reporting new transactions to the wormhole contract
+	Watcher struct {
+		aptosRPC     string
+		aptosAccount string
+		aptosHandle  string
+
+		msgChan  chan *common.MessagePublication
+		obsvReqC chan *gossipv1.ObservationRequest
+	}
+)
+
+var (
+	aptosMessagesConfirmed = promauto.NewCounter(
+		prometheus.CounterOpts{
+			Name: "wormhole_aptos_observations_confirmed_total",
+			Help: "Total number of verified Aptos observations found",
+		})
+	currentAptosHeight = promauto.NewGauge(
+		prometheus.GaugeOpts{
+			Name: "wormhole_aptos_current_height",
+			Help: "Current Aptos block height",
+		})
+)
+
+// NewWatcher creates a new Aptos appid watcher
+func NewWatcher(
+	aptosRPC string,
+	aptosAccount string,
+	aptosHandle string,
+	messageEvents chan *common.MessagePublication,
+	obsvReqC chan *gossipv1.ObservationRequest,
+) *Watcher {
+	return &Watcher{
+		aptosRPC:     aptosRPC,
+		aptosAccount: aptosAccount,
+		aptosHandle:  aptosHandle,
+		msgChan:      messageEvents,
+		obsvReqC:     obsvReqC,
+	}
+}
+
+func (e *Watcher) retrievePayload(s string) ([]byte, error) {
+	res, err := http.Get(s) // nolint
+	if err != nil {
+		return nil, err
+	}
+	body, err := ioutil.ReadAll(res.Body)
+	if err != nil {
+		return nil, err
+	}
+	return body, err
+}
+
+func (e *Watcher) observeData(logger *zap.Logger, data gjson.Result, native_seq uint64) {
+	em := data.Get("sender")
+	if !em.Exists() {
+		logger.Info("sender")
```

Comment:
Should this message be an error or more informative?

Reply:
Fixed